File: Arrays & Strings/#151- Reverse words in a String - Medium/reverseWords.cpp
class Solution {
public:
    string reverseWords(string s) {
        int left = 0;
        int right = s.length()-1;
    
        string temp="";
        string ans="";
    
        //Iterate the string and keep on adding to form a word
        //If empty space is encountered then add the current word to the result
        while (left <= right) {
            char ch= s[left];
            if (ch != ' ') {
                temp += ch;
            } else if (ch == ' ') {
                if (ans!="") ans = temp + " " + ans;
                else ans = temp;
                temp = "";
            }
            left++;
        }
    
        //If not empty string then add to the result(Last word is added)
        if (temp!="") {
            if (ans!="") ans = temp + " " + ans;
            else ans = temp;
        }
    
        return ans; 
    }
};

File: Arrays & Strings/#151- Reverse words in a String - Medium/reverseWords.py
class Solution:
    def reverseWords(self, s: str) -> str:
        left = 0
        right = len(s) - 1

        temp = ""
        ans = ""

        # Skip leading spaces
        while left <= right and s[left] == ' ':
            left += 1
        # Skip trailing spaces
        while right >= left and s[right] == ' ':
            right -= 1

        # Iterate the string and collect words
        while left <= right:
            ch = s[left]
            if ch != ' ':
                temp += ch
            else:
                if temp != '':
                    if ans != '':
                        ans = temp + " " + ans
                    else:
                        ans = temp
                    temp = ""
            left += 1

        # Add the last word if any
        if temp != "":
            if ans != "":
                ans = temp + " " + ans
            else:
                ans = temp

        return ans
